bootstrap/templates/k0s-config.yaml.j2

```
@@ -0,0 +1,118 @@
+#jinja2: trim_blocks: True, lstrip_blocks: True
+---
+apiVersion: k0sctl.k0sproject.io/v1beta1
+kind: Cluster
+metadata:
+  name: k0s-cluster
+spec:
+  hosts:
+  {% for item in bootstrap_nodes.master %}
+  {% if (item.role is not defined) and (bootstrap_nodes.master | length > 1) %}
+  - role: controller+worker
+  {% else %}
+  - role: {{ item.role | default('controller') }}
+  {% endif %}
+    ssh:
+      address: {{ item.address }}
+      user: {{ item.username }}
+    installFlags:
+    - --disable-components=metrics-server
+    {% if ((item.role is not defined) and (bootstrap_nodes.master | length > 1)) or (item.role | default('') == 'controller+worker') %}
+    - --no-taints
+    {% endif %}
+  {% endfor %}
+{% if bootstrap_nodes.worker | default([]) | length > 0 %}
+  {% for item in bootstrap_nodes.worker %}
+  - role: {{ item.role | default('worker') }}
+    ssh:
+      address: {{ item.address }}
+      user: {{ item.username }}
+  {% endfor %}
+{% endif %}
+  k0s:
+    version: v1.28.4+k0s.0
```

```suggestion
    # renovate: datasource=github-releases depName=k0sproject/k0s
    version: "v1.28.4+k0s.0"
```

Quotes matter on this value.

---

Also renovate change needed:

```json
    // custom versioning
    {
      "description": "Use custom versioning for k0s/k3s",
      "matchDatasources": ["github-releases"],
      "versioning": "regex:^v(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)(?<compatibility>\\+k?s)(?<build>\\d+)$",
      "matchPackagePatterns": ["k0s", "k3s"]
    },
```